### ISSUE TITLE
Sometimes opens two tabs in combination with HTTPS Everywhere & Redirector

### ISSUE THREAD

**Issue by stoically** [stoically/firefox-add-on-temporary-containers#43]
Under some rare circumstances it happens that in combination with [HTTPS Everywhere](https://addons.mozilla.org/en-US/firefox/addon/https-everywhere) two tabs open upon navigation. Disabling and enabling the HTTPS Everywhere Add-on in that case helps. I couldn't identify the underlying problem here yet.

Maybe related #42? (cc @anewuser)

**Comment by anewuser** [comment]
I don't have HTTPS Everywhere but do use [Redirector](https://addons.mozilla.org/firefox/addon/5064) with a rule to redirect some requests to HTTPs, including the domain I experienced this bug with (`archive.org`).

**Comment by stoically** [comment]
Ok, so in the cases where this happens the `onBeforeRequest` handler sees two requests

```[browser.webRequest.onBeforeRequest] incoming request Object { requestId: "30521", url: "http://npmjs.org/", originUrl: undefined, documentUrl: undefined, method: "GET", type: "main_frame", timeStamp: 1518055600517, frameId: 0, parentFrameId: -1, proxyInfo: null, 3 more… }  background.js:82:17
[browser.webRequest.onBeforeRequest] incoming request Object { requestId: "30521", url: "https://npmjs.org/", originUrl: undefined, documentUrl: undefined, method: "GET", type: "main_frame", timeStamp: 1518055600544, frameId: 0, parentFrameId: -1, proxyInfo: null, 3 more… }  background.js:82:17```

This doesn't happen always and they have the same requestId (only the protocol is different). So I'm really curious as of way and how this only happens sometimes and why it has the same requestId.

**Comment by stoically** [comment]
Ok, so it turns out for some reason HTTPS Everywhere makes a new request despite Temporary Containers canceling it first, that's why the two requests show up. Added a test to check whether a requests opened a tab already and if so, don't open further tabs. The same requestId comes in handy.

Just published version 0.37 with the fix. Works hopefully for Redirector too. Let me know if it pops up again.
